examples/java-matter-controller/java/src/com/matter/controller/commands/common/Command.java

```diff
@@ -0,0 +1,243 @@
+/*
+ *   Copyright (c) 2022 Project CHIP Authors
+ *   All rights reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ *
+ */
+
+package com.matter.controller.commands.common;
+
+import java.util.ArrayList;
+import java.util.Optional;
+
+public abstract class Command {
+  private String mName = null;
+  private String mHelpText = null;
+
+  private boolean mIsInteractive = false;
+  private ArrayList<Argument> mArgs = new ArrayList<Argument>();
+
+  public Command(String commandName) {
+    this.mName = commandName;
+  }
+
+  public Command(String commandName, String helpText) {
+    this.mName = commandName;
+    this.mHelpText = helpText;
+  }
+
+  public final String getName() {
+    return mName;
+  }
+
+  public final String getHelpText() {
+    return mHelpText;
+  }
+
+  public final String getAttribute() {
+    for (int i = 0; i < mArgs.size(); i++) {
+      Argument arg = mArgs.get(i);
+      if (arg.type == ArgumentType.Attribute) {
+        return (String) arg.value;
+      }
+    }
+
+    return null;
+  }
+
+  public final String getArgumentName(int index) {
```

Comment:
annotate with nullable. Why are we returning null instead of just using get and accepting the exception if an invalid index is used?

Reply:
Same comment for items bewlow.